src/telemetry/CMakeLists.txt

```diff
@@ -26,3 +26,10 @@ if (TELEMETRY_INSTALL_TARGETS)
 	install(TARGETS telemetry LIBRARY DESTINATION ${INSTALL_DIR_LIB})
 	install(DIRECTORY ${PROJECT_SOURCE_DIR}/include/ DESTINATION ${INSTALL_DIR_INCLUDE})
 endif()
+
+if (TELEMETRY_ENABLE_TESTS)
+	add_executable(testTelemetry ${TELEMETRY_SOURCE_FILES})
+	target_compile_definitions(testTelemetry PRIVATE TELEMETRY_ENABLE_TESTS)
+	target_link_libraries(testTelemetry telemetry GTest::GTest)
+	add_test(testTelemetry testTelemetry)
```

Comment:
Replace `add_test` with following for better integration with CTest:
```
include(GoogleTest)
gtest_discover_tests(testTelemetry)
```

Reply:
Updated